kong/plugins/basic-auth/basicauth_credentials.lua

```diff
@@ -27,7 +27,11 @@ local hash_password = function(self, cred_id_or_username, cred)
     end
   end
 
-  cred.password = crypto.hash(cred.consumer.id, cred.password)
+  if cred.consumer then
+    cred.password = crypto.hash(cred.consumer.id, cred.password)
+  else
+    cred.password = crypto.hash(utils.uuid(), cred.password)
```

Comment:
If `cred.consumer` is required, is this case ever going to get hit? (It seems odd to hash something based on a random value, how would this hash ever get verified?)

Reply:
@hishamhm, that is a good question, I didn't like to add that, but the hashing seems to happen prior validation (which sucks). And that's why I added that. The validation will still fail, it will just not cry about password being required.